src/components/CardGridSearch.astro

```diff
@@ -0,0 +1,255 @@
+---
+/* CSS and html structure is based on:
+ * https://github.com/CloudCannon/pagefind/blob/main/pagefind_ui/default/svelte/ui.svelte
+ * And CSS variables from Starlight from:
+ * https://github.com/withastro/starlight/blob/a0f6fa80d9be1fbe85530323a0b2523acb83a15a/packages/starlight/components/Search.astro
+ */
+
+interface Props {
+	placeholder?: string;
+	title?: string;
+}
+const { placeholder = 'Quick search a feature...', title = 'Search a feature' } = Astro.props;
+
+// TODO: Expose placeholder and title to i18n
+---
+
+<card-search>
+	<div id="search-input" class="pagefind-ui pagefind-ui--reset pagefind-ui--reset">
```

Comment:
Duplicate css of `pagefind-ui--reset`, but this also doesn't seem to be defined below. You may be looking for something like [`.not-content`](https://starlight.astro.build/guides/components/#compatibility-with-starlights-styles)?
```suggestion
	<div id="search-input" class="pagefind-ui pagefind-ui--reset">
```

Reply:
`.pagefind-ui--reset` comes from https://github.com/CloudCannon/pagefind/blob/main/pagefind_ui/default/svelte/reset.svelte I can probably copy over some styles so we don't have this hidden dependency. What do you think? That's why I left pagefind-ui prefix in all classes